pkg/plugins/autodiscovery/flux/helmrepository.go

```diff
@@ -0,0 +1,66 @@
+package flux
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"gopkg.in/yaml.v3"
```

Comment:
You could use `metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"` here and `"sigs.k8s.io/yaml"`

Which would let you use `metav1.TypeMeta` instead of extracting the `ApiVersion` and `Kind` here?

This would then let you do something like...

```go
gvk := helmRepository.GroupVersionKind()
if gvk.GroupKind().String() == "HelmRepository.source.toolkit.fluxcd.io" {
        return &helmRepository, nil
}
```
Which would tighten down the criteria for `HelmRepository` resources in the repo?

Bearing in mind that these are not coming from the cluster, so...there could be invalid resources in-repo that would not be applied to the cluster.

Reply:
I admit I hesitated doing that for other plugins as well, my main motivation for not doing it was to reduce the number of Golang dependencies needed for Updatecli considering I could just create the struct I need.

Reply:
Regarding `sigs.k8s.io/yaml` vs `gopkg.in/yaml.v3` if I recall correctly the main motivation of `sigs.k8s.io/yaml` is being able to use the same struct tag for both json and yaml files because otherwise sigs.k8s.io is based on `gopkg.in/yaml.v3` and so share the same limitation

Reply:
One benefit of the sigs.k8s.io approach is that it's parsing it as it would be parsed by K8s.

Reply:
Thanks for the suggestion, I switched to `sigs.k8s.io/yaml` in futur, I may applied the same improvement to other Kubernetes related plugins